src/imitation/rewards/reward_nets.py

```diff
@@ -700,35 +715,77 @@ def predict_reward_moments(
             action: Actions of shape `(batch_size,) + action_shape`.
             next_state: Successor states of shape `(batch_size,) + state_shape`.
             done: End-of-episode (terminal state) indicator of shape `(batch_size,)`.
+            **kwargs: passed along to predict processed.
 
         Returns:
-            Computed reward std of shape `(batch_size,)`.
+            * Reward mean of shape `(batch_size,)`.
+            * Reward std of shape `(batch_size,)`.
         """
-        state, action, next_state, done = self.preprocess(
+        batch_size = state.shape[0]
+        all_rewards = self.predict_processed_all(
             state,
             action,
             next_state,
             done,
+            **kwargs,
         )
+        mean_reward = all_rewards.mean(-1)
+        var_reward = all_rewards.var(-1, ddof=1)
+        assert mean_reward.shape == var_reward.shape == (batch_size,)
+        return mean_reward, var_reward
+
+    def forward(self, *args) -> th.Tensor:
+        """The forward method of the ensemble should in general not be used directly."""
+        raise NotImplementedError()
+
+    def predict_processed(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+        **kwargs,
+    ) -> np.ndarray:
+        """Return the mean of the ensemble members."""
+        mean, _ = self.predict_reward_moments(state, action, next_state, done, **kwargs)
+        return mean
 
-        all_rewards = self.forward_all(state, action, next_state, done)
-        return all_rewards.mean(-1).cpu().numpy(), all_rewards.var(-1).cpu().numpy()
+    def predict(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+        **kwargs,
+    ):
+        """Return the mean of the ensemble members."""
+        return self.predict_processed(state, action, next_state, done, **kwargs)
```

Comment:
It seems marginally more natural to me to have predict_processed call predict than the other way around (as if they're equal then the return value is presumably "unprocessed").